adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotations.scala

```diff
@@ -0,0 +1,140 @@
+    "I2" -> INFO_COMPOUND_ANNOTATION,
+    "I3" -> INFO_NON_REFERENCE_ANNOTATION
+  )
+
+  private def parseEffects(s: String): List[String] = {
+    s.split("&").toList
+  }
+
+  private def parseMessages(s: String): List[VariantAnnotationMessage] = {
+    // todo: haven't seen a delimiter here, assuming it is also '&'
+    s.split("&").map(MESSAGES.get(_)).toList.flatten
+  }
+
+  private def parseFraction(s: String): (Option[Integer], Option[Integer]) = {
+    if ("".equals(s)) {
+      return (None, None)
+    }
+    val tokens = s.split("/")
+    tokens.length match {
+      case 0 => (None, None)
+      case 1 => (Some(Integer.parseInt(tokens(0))), None)
+      case _ => (Some(Integer.parseInt(tokens(0))), Some(Integer.parseInt(tokens(1))))
+    }
+  }
+
+  private def notEmpty(s: String): Boolean = Option(s) match {
+    case None    => false
+    case Some(s) => s.length > 0
+  }
+
+  private[converters] def parseTranscriptEffect(s: String): Seq[TranscriptEffect] = {
+    val fields = s.split("\\|", -1)
+    if (fields.length < 16) {
+      if (log.isWarnEnabled) log.warn("Invalid VCF ANN attribute, expected at least 16 fields, found {}: {}", fields.length, fields)
+      return Seq()
+    }
+
+    // simple and ignored fields
+    val (alternateAllele, annotationImpact, geneName, geneId, featureType,
+      featureId, biotype, transcriptHgvs, proteinHgvs, distance) =
+      (fields(0), fields(2), fields(3), fields(4), fields(5),
+        fields(6), fields(7), fields(9), fields(10), fields(14))
+
+    // multivalued fields
+    val effects = parseEffects(fields(1))
+    val (rank, total) = parseFraction(fields(8))
+    val (cdnaPosition, cdnaLength) = parseFraction(fields(11))
+    val (cdsPosition, cdsLength) = parseFraction(fields(12))
+    val (proteinPosition, proteinLength) = parseFraction(fields(13))
+    val messages = parseMessages(fields(15))
+
+    val te = TranscriptEffect.newBuilder()
+    if (notEmpty(alternateAllele)) te.setAlternateAllele(alternateAllele)
+    if (!effects.isEmpty) te.setEffects(effects.asJava)
+    if (notEmpty(geneName)) te.setGeneName(geneName)
+    if (notEmpty(geneId)) te.setGeneId(geneId)
+    if (notEmpty(featureType)) te.setFeatureType(featureType)
+    if (notEmpty(featureId)) te.setFeatureId(featureId)
+    if (notEmpty(biotype)) te.setBiotype(biotype)
+    rank.foreach(te.setRank(_))
+    total.foreach(te.setTotal(_))
+    if (notEmpty(transcriptHgvs)) te.setTranscriptHgvs(transcriptHgvs)
+    if (notEmpty(proteinHgvs)) te.setProteinHgvs(proteinHgvs)
+    cdnaPosition.foreach(te.setCdnaPosition(_))
+    cdnaLength.foreach(te.setCdnaLength(_))
+    cdsPosition.foreach(te.setCdsPosition(_))
+    cdsLength.foreach(te.setCdsLength(_))
+    proteinPosition.foreach(te.setProteinPosition(_))
+    proteinLength.foreach(te.setProteinLength(_))
+    if (notEmpty(distance)) te.setDistance(Integer.parseInt(distance))
+    if (!messages.isEmpty) te.setMessages(messages.asJava)
+
+    Seq(te.build())
```

Comment:
Same pattern used in feature parsing, `TranscriptEffect` association is an array in `VariantAnnotation` schema and a `java.util.List<TranscriptEffect>` in the Avro-generated object.